docs/14_0_0/components/tree.md

```diff
@@ -124,12 +124,79 @@ and tree is created, this mode is suitable for relatively small datasets and pro
 interaction. On the otherhand it’s not suitable for large data since all the data is sent to the client also
 client side tree is stateless.
 
-_Dynamic:_ Dynamic mode uses ajax to fetch the treenodes from server side on demand, compared to
+_Dynamic:_ Dynamic mode uses AJAX to fetch the treenodes from server side on demand, compared to
 the client toggling, dynamic mode has the advantage of dealing with large data because only the
 child nodes of the root node is sent to the client initially and whole tree is lazily populated. When a
 node is expanded, tree only loads the children of the particular expanded node and send to the client
 for display.
 
+## Lazy Loading
+As extension to the dynamic tree, we also offer the `LazyTreeNode` model, which enables lazy loading of child nodes.  
+It takes 2 functions: 
+- one to load the child data
+- one to decide if the current data has children or not (leaf)
+
+```xhtml
+<p:tree value="#{treeBean.root}" var="node">
+    <p:treeNode>
+        <h:outputText value="#{node}"/>
+    </p:treeNode>
+</p:tree>
+```
+```java
+@Named
+@ViewScoped
+public class TreeBean implements Serializable {
+
+    private TreeNode<FileInfo> root;
+
+    @PostConstruct
+    public void init() {
+        FacesContext context = FacesContext.getCurrentInstance();
+        root = new LazyTreeNode<>(new FileInfo(context.getExternalContext().getRealPath("/"), true),
+                (fileInfo) -> listFiles(fileInfo),
+                (fileInfo) -> !fileInfo.isDirectory());
```

Comment:
no parenthesis for one argument (otherwise IDE complain about it

Reply:
my IDE doesnt complain, so just cosmetics and we didnt have a checkstyle rule for it